eiffel-vocabulary/EiffelIssueDefinedEvent.md

```diff
@@ -0,0 +1,278 @@
+<!---
+   Copyright 2018 Jaden Young
+   For a full list of individual contributors, please see the commit history.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+--->
+
+# EiffelIssueDefinedEvent (ID)
+The EiffelIssueDefinedEvent declares that an issue has been created in some
+external issue management software. This event provides provenance for an issue,
+and acts as a "handle" for future issue related events similarly to
+[EiffelActivityTriggeredEvent](./EiffelActivityTriggeredEvent.md) and
+[EiffelTestCaseTriggeredEvent](./EiffelTestCaseTriggeredEvent.md). It provides
+a minimal amount of information necessary for this purpose and allows further
+data to be attached through later events, for example the 
+[EiffelIssueAssignedEvent](./EiffelIssueAssignedEvent).
+
+## Data Members
+
+### data.type
+__Type:__ String  
+__Required:__ Yes  
+__Legal values:__ BUG, IMPROVEMENT, FEATURE, WORK_ITEM, REQUIREMENT, OTHER  
+__Description:__ The type of issue.  
+
+### data.tracker
+__Type:__ String  
+__Required:__ Yes  
+__Description:__ The name of the issue tracker. This can unfortunately not be
+standardized, and is therefore context sensitive: though some trackers and ALM
+tools are more popular than others, an exhaustive enumeration is impossible,
+particularly when considering company specific internal solutions. Consequently
+one should not rely on the name as the primary method of retrieval, but rather
+__data.uri__. __data.tracker__ together with __data.id__
+is still useful for analysis and traceability, however, as long as it can be
+correctly interpreted.
+
+### data.id
+__Type:__ String  
+__Required:__ Yes  
+__Description:__ The identity of the issue. This is tracker dependent - most
+trackers have their own issue naming schemes.  
+
+### data.uri
+__Type:__ String  
+__Required:__ Yes  
+__Description:__ A URI that links to a document describing the issue in depth.
+
+### data.title
+__Type:__ String  
+__Required:__ No  
+__Description:__ A one-line title or summary of the issue. This exists mostly
+for human consumption, as "Implement widget X" is much more meaningful to a
+human when viewing a graph of Eiffel events than "1302". This is not meant
+to be a detailed description, as such information should be accessible by
+following __data.uri__.
+
+### data.initialCategory
```

Comment:
See my comment on this field in the ISM event.
Furthermore, the description talks about 'state' but this field is called 'category'. I guess I would prefer 'initialState' here. 'category' is what I would use for the issue type instead...

Reply:
The naming of these fields is a little unwieldly. The idea behind the category/status divide on ISM was to have a closed set of "something that needs work", "something being worked on", and "something that doesn't need any more work" in `category`, and leave any specific subclasses of those up to `status`. This would give consumers a simple and well defined path to react to/filter on the high level changes in `category` without having to worry about the underlying issue management software but still allow the flexibility of having a custom status. "TODO", "DOING", and "DONE" would probably have been better choices for categories, but we didn't want to insinuate that we were forcing kanban style/terminology. The semantics of `category` relied a bit on the context of being inside of an IssueStatusModifiedEvent, so `initalCategory` sounds a bit funny in an ID. We were thinking of ID/ISM being closer to ActT/ActS, where an issue defined with a status/category would be emitted in Eiffel as ID immediately followed by ISM.

As for how the event should be consumed, while we should definitely keep consumers' needs in mind, [I would be cautious of being perscriptive](https://github.com/jaden-young/eiffel/blob/master/introduction/introduction.md#how-does-eiffel-achieve-flexibility).

Reply:
Since we're moving to get the refactoring of SCC/IV into ID done first, I've removed initalCategory/initalStatus as they don't really make much sense without ISM existing.